cmd/engine/engine.go

```diff
@@ -410,12 +345,37 @@
 		}()
 	}
 
-	// serve the gRPC socket
-	if err := server.Serve(sock); err != nil {
-		log.Infof("Failed to serve gRPC endpoint: %s", err)
+	log.Infof("Starting gRPC endpoint on :%d", grpcPort)
+
+	// Start the gRPC server
+	_, srv, err = server.StartGRPCServer(
```

Comment:
Maybe adding comment to the `StartGRPCServer` function that it is somehow asynchronous/non-blocking? At a first sight, I thought the gateway below won't be started since starting grpc server is blocking maybe